text/accumulation.tex

```diff
@@ -30,118 +34,206 @@ \subsection{Preimage Integration}
+\begin{align}
+  \accumulated &\in \seq{\dict{\H}{\H}}_\mathsf{E} \\
+  \wideparen{\accumulated} &\equiv \{ x \mid x \in \accumulated \}
+\end{align}
 
-We finally define the results which will be given as an operand into the accumulate function for each service in $\mathbf{S}$. This is a sequence of operand tuples $\mathbb{O}$, one sequence for each service in $\mathbf{S}$. Each sequence contains one element per work output (or error) to be accumulated for that service, together with said work output's payload hash, package hash and authorization output. The tuples are sequenced in the same order as they appear in $\mathbf{W}$. Formally:
+We also maintain knowledge of available but not-yet-accumulated work-reports in the state item $\ready$. Each of these have been available for at most one epoch but have or had unfulfilled dependencies. Alongside the work-report itself, we retain its unaccumulated dependencies, a set of work-package hashes. Formally:
 \begin{align}
-  \mathbb{O} \equiv \ltuple\isa{o}{\Y \cup \mathbb{J}}\ts\isa{l}{\H}\ts\isa{k}{\H}\ts\isa{a}{\Y}\rtuple
+  \ready &\in \seq{\seq{(\mathbb{W}, \{\H\})}}_\mathsf{E}
 \end{align}
+
+The newly available work-reports, $\mathbf{W}$, are partitioned into two sequences based on the condition of having zero prerequisite work-reports. Those meeting the condition, $\mathbf{W}^!$, are accumulated immediately. Those not, $\mathbf{W}^Q$, are for queued execution. Formally:
 \begin{align}
-  M\colon\left\{\ \begin{aligned}
-    \N_S &\to \lseq\mathbb{O}\rseq \\
-    \displaystyle s &\mapsto \left[ \ltup\begin{alignedat}{3}
-        \is{o&}{\mathbf{r}_o}\ts\is{&l&}{\mathbf{r}_p}\ts\\
-        \is{a&}{w_o}\ts\is{&k&}{(w_s)_h}
-    \end{alignedat}\rtup
-      \,\middle\vert\ 
-    \begin{aligned}
-      w &\in \mathbf{W},\\
-      \mathbf{r} &\in w_\mathbf{r},\\
-      \mathbf{r}_s &= s
-    \end{aligned}\ \right]
-  \end{aligned}\right.
+  \mathbf{W}^! &\equiv [ w \mid w \orderedin \mathbf{W}, (w_x)_p = \none \wedge w_\wr¬srlookup = \{\} ] \\
+  \mathbf{W}^Q &\equiv E(\sq{
+    D(w) \mid
+    w \orderedin \mathbf{W},
+    (w_x)_p \ne \none \vee w_\wr¬srlookup \ne \{\}
+  }, \wideparen{\accumulated})\!\!\\
+  D(w) &\equiv (w, (w_x)_p \cup \keys{w_\wr¬srlookup})
 \end{align}
 
-\subsection{Invocation}
+We must define the two dependent functions. Firstly, the mapping function $\srmap$ which builds a dictionary of work-package hashes to segment-roots from a set of work-reports. Formally:
```

Comment:
Wording here is a bit confusing as it sounds like `\srmap` has already been used but AFAICT it is only used in the definition of Q below?